File: linear-search.js

/**
 * implement linear search
 * traverse the array and return position of the element that matches search query
 * 
 */

function linearSearch(arr,item, position){
  for(position = 0;position<arr.length; position++) {
    if(arr[position] == item) {
      console.log("match found at position %d", position+1);
      return position+1;
    }
  }
  console.log("no match found..\n");
}

linearSearch([10,20,3,6,8,5,44,33,55,33,54,67,21,37],5,0);

File: search/linear-search.c
#include <stdio.h>
#include "stdlib.h"
#include "string.h"
int main(void) {
  int list[14] = {10,20,3,6,8,5,44,33,55,33,54,67,21,37};
  int position,item, i;

  /**
  -linear search is the simplest of all search algorithms
  -involves navigating linearly across whole list and returning the position where an item has found
  */
  while(1){
  printf("Enter item to search:\n");
  scanf("%d",&item);
  for(i =0; i<14;i++) {
    if(list[i] == item) {
      printf("match found at position %d\n",i+1);
      return i+1;
    }
  }
    printf("No match found\n");
  }

  return 0;
}